addons/mrp/models/mrp_production.py

```diff
@@ -1974,17 +1972,27 @@ def action_merge(self):
         production._create_workorder()
 
         for move in production.move_raw_ids:
-            move.move_orig_ids = [Command.set(origs[move.bom_line_id.id])]
+            for field, vals in origs[move.bom_line_id.id].items():
+                move[field] = vals
         for move in production.move_finished_ids:
             move.move_dest_ids = [Command.set(dests[move.byproduct_id.id])]
-        production.move_dest_ids = [Command.set(sum(list(dests.values()), []))]
+
+        for p in self:
+            if p.move_dest_ids:
+                p.move_dest_ids.created_production_id = production.id
 
         self.procurement_group_id.stock_move_ids.group_id = production.procurement_group_id
 
         if 'confirmed' in self.mapped('state'):
-            production.action_confirm()
+            moves = (production.move_raw_ids | production.move_dest_ids)
+            moves._adjust_procure_method()
+            moves.write({'state': 'confirmed'})
```

Comment:
Not sure why we need `move_dest_ids` here. And I guess we also need reservation here?

Reply:
merged/split moves could only be draft or confirm so no need for reservation IMO. Also we mark the move_finished_ids as 'confirmed' if the initial MO's were also confirm in order to avoid a second trigger of push rule

Reply:
so it should be `move_finished_ids` instead of `move_dest_ids`

Reply:
Oups. You are correct :+1: